src/Raven.Client/ServerWide/Tcp/TcpNegotiation.cs

```diff
@@ -73,11 +73,9 @@ await using (var writer = new AsyncBlittableJsonTextWriter(context, stream))
                 }
 
                 var supportedFeatures = TcpConnectionHeaderMessage.GetSupportedFeaturesFor(parameters.Operation, current);
-
-                return new TcpConnectionHeaderMessage.SupportedFeatures(supportedFeatures)
-                {
-                    DataCompression = dataCompression
-                };
+                var results = new TcpConnectionHeaderMessage.SupportedFeatures(supportedFeatures);
+                results.DataCompression &= dataCompression;
+                return results;
```

Comment:
This is the fix that was already merged.
So please rebase this PR

Reply:
@shaharhikri ping

Reply:
Done